### ISSUE TITLE
Nothing useful happens after esy run.

### ISSUE THREAD

**Issue by m-kru** [onivim/oni2#808]
I have followed [Build and Run](https://onivim.github.io/docs/for-developers/building) to run Oni2, but I have problem with `esy run` command. When I execute `esy run` I can only see single rectangle blink in the middle of the screen. Neither error nor message is printed in the terminal.

**Comment by CrossR** [comment]
Hi! Sorry for the issues caused!


Could you run `esy run -f --checkhealth`? That might give us some extra info.
Also running `esy run -f` (No fork, which sends the debug info to the screen) whilst you have the `ONI2_DEBUG` environment variable set to `1`?

That should let us start seeing some of the debug information, to work out what is causing the issue here.

**Comment by m-kru** [comment]
`esy run -f --checkhealth` is ok, everything passes.

`esy run -f` output:
```
[DEBUG] Startup: Parsing CLI options complete
[DEBUG] Starting Onivim 2.
[DEBUG] Calling App.start
[DEBUG] Init
[INFO] [Revery]: Window: Starting window creation...
[INFO] [Revery]: Window: Using vsync: true
[INFO] [Revery]: Window: Creating window Oni2 width: 800 height: 600
[INFO] [Revery]: Window: Window created - id: 2
[INFO] [Revery]: Window: Setting window context
X Error of failed request:  GLXBadDrawable
  Major opcode of failed request:  152 (GLX)
  Minor opcode of failed request:  5 (X_GLXMakeCurrent)
  Serial number of failed request:  251
  Current serial number in output stream:  251
```

**Comment by CrossR** [comment]
Hm, that looks to be the same error as over here: https://github.com/revery-ui/revery/issues/568

Are you on Arch as well?

**Comment by m-kru** [comment]
Yes, I use Arch.

**Comment by bryphe** [comment]
Bummer! Looks like @glennsl isn't the only one hitting this with the upgrade to SDL2 https://github.com/revery-ui/revery/issues/568

Just for extra data:
- What type of GPU are you using, @m-kru ? I'm wondering if this is specific to the newer NVidia drivers - or something more general.

**Comment by m-kru** [comment]
```
01:00.0 VGA compatible controller: NVIDIA Corporation GP106M [GeForce GTX 1060 Mobile] (rev a1)
	Subsystem: Tongfang Hongkong Limited GP106M [GeForce GTX 1060 Mobile]
	Kernel driver in use: nvidia
	Kernel modules: nouveau, nvidia_drm, nvidia

```

**Comment by necabo** [comment]
Hitting this as well on arch + NVIDIA card:

```
01:00.0 VGA compatible controller: NVIDIA Corporation GM204 [GeForce GTX 980] (rev a1)
	Subsystem: NVIDIA Corporation GM204 [GeForce GTX 980]
	Kernel driver in use: nvidia
	Kernel modules: nouveau, nvidia_drm, nvidia
```
`git-bisect`ing it indeed points to the SDL2 integration (`f459914ea0dd0830aa6945420acb38958c5a896f`), commits before that still work just fine.

**Comment by bryphe** [comment]
Thanks @necabo and @m-kru for the details and the investigation.

I debugged this with @glennsl earlier today - looks like it was fixed for him in this commit: https://github.com/revery-ui/reason-sdl2/commit/9e93f9d565e2adce23635f839aa014f93999c0ec

I've just pulled in that change with #813 - would you mind trying on the latest `master`?

**Comment by m-kru** [comment]
Ok, now I am able to launch Oni2. But to be honest, the number of bugs, that I have spotted in 5 minutes is surprisingly high :( .

**Comment by CrossR** [comment]
>Ok, now I am able to launch Oni2. But to be honest, the number of bugs, that I have spotted in 5 minutes is surprisingly high :( .

We are still in pre-alpha! Bugs are slowlgy getting ticked off, I think mainly based on the number of 👍 s the issues on here have. Because of that, its useful to check if there is an issue/raise one.

**Comment by necabo** [comment]
> I've just pulled in that change with #813 - would you mind trying on the latest `master`?

Can confirm it "working for me"™ again.

**Comment by bryphe** [comment]
Thanks for confirming that it launches now with #813 , @necabo and @m-kru !

> Ok, now I am able to launch Oni2. But to be honest, the number of bugs, that I have spotted in 5 minutes is surprisingly high :( .

Yes, we have our work cut out for us! But logging them is helpful - thank you 👍  And '+1'ing issues that impact you helps us prioritize.

I'll close this particular issue now as it was resolved with #813
